### PR TITLE
gh-90997: bpo-46841: Disassembly of quickened code

### PR DESCRIPTION
**Pull request by penguin-wwy**
https://bugs.python.org/issue46841

### CODE REVIEW

**Comment by brandtbucher** [comment]
Is this ready for review?

If not, perhaps mark it as a "draft" for now.

File: Lib/opcode.py [hunk 1]
```diff
@@ -336,6 +336,17 @@ def jabs_op(name, op, entries=0):
 _specialized_instructions = [
     opcode for family in _specializations.values() for opcode in family
 ]
+
+_empty_slot = [slot for slot, name in enumerate(opname) if name.startswith("<")]
+for spec_op, specialized in zip(_empty_slot, _specialized_instructions):
```
Comment (markshannon) [inline Lib/opcode.py]: I don't think that we want the specialized forms in `opmap` and `opname`, as they are internal and likely to change between micro versions.

Could you move this code to the dis module?

**Comment by markshannon** [comment]
Looks promising. Will need some tests for the new output format.

File: Lib/opcode.py (hunk 1 above)
Comment (penguin-wwy) [reply Lib/opcode.py]: Finish :)

**Comment by penguin-wwy** [comment]
> Looks promising. Will need some tests for the new output format.

Already add some test case for quicken code in the dis.

File: Lib/test/test_dis.py [hunk 2]
```diff
@@ -863,6 +887,79 @@ def check(expected, **kwargs):
         check(dis_nested_2, depth=None)
         check(dis_nested_2)
 
+    _testinternalcapi = import_helper.import_module('_testinternalcapi')
+
+    @cpython_only
+    def test_super_instructions(self):
+        self._testinternalcapi.code_quicken(load_test.__code__)
```
Comment (markshannon) [inline Lib/test/test_dis.py]: I'd rather not have to add the extra `_testinternalcapi` function.
`[load_test(0,0) for _ in range(100)]` would be better, as it reflects the way code will get quickened normally.

Comment (markshannon) [reply Lib/test/test_dis.py]: Or, given that it gets used a few times, just add a little helper function to run the function 100 times.

**Review by markshannon — COMMENTED** [review]
Could you remove the changes to `_testinternalcapi`, so we don't need to export the `_PyCode_Quicken` function.

File: Lib/test/test_dis.py (hunk 2 above)
Comment (penguin-wwy) [reply Lib/test/test_dis.py]: Finish

**Comment by penguin-wwy** [comment]
I have made the requested changes; please review again :) @markshannon @brandtbucher

File: Lib/test/test_dis.py [hunk 3]
```diff
@@ -860,6 +912,93 @@ def check(expected, **kwargs):
...
+        self.code_quicken(lambda: exec(co_int, {}, {'a': 1, 'b': 2}))
+        got = self.get_disassembly(co_int, adaptive=True)
+        self.do_disassembly_compare(got, binary_op_quicken % "BINARY_OP_ADD_INT                 0 (+)", True)
+
+        co_unicode = compile('a + b', "<unicode>", "eval")
+        self.code_quicken(lambda: exec(co_unicode, {}, {'a': 'a', 'b': 'b'}))
+        got = self.get_disassembly(co_unicode, adaptive=True)
+        self.do_disassembly_compare(got, binary_op_quicken % "BINARY_OP_ADD_UNICODE             0 (+)", True)
+
+        binary_subsrc_quicken = """\
```
Comment (markshannon) [inline Lib/test/test_dis.py]:
```suggestion
        binary_subscr_quicken = """\
```

**Comment by markshannon** [comment]
On small typo, otherwise looks good.

**Comment by penguin-wwy** [comment]
Thanks, I have made the requested changes.

**Comment by penguin-wwy** [comment]
Thanks. :)
I have made the requested changes; please review again

**Comment by markshannon** [comment]
Thanks @penguin-wwy for doing this.

**Pull request merged**